lgc/patch/NggPrimShader.cpp

```diff
@@ -2484,24 +2499,16 @@ void NggPrimShader::loadStreamOutBufferInfo(Value *userData) {
 
     Value *ptr = m_builder.CreateInsertElement(pc, ptrValue, static_cast<uint64_t>(0));
     ptr = m_builder.CreateBitCast(ptr, m_builder.getInt64Ty());
-    ptr = m_builder.CreateIntToPtr(ptr, ptrTy, "globalTablePtr");
+    ptr = m_builder.CreateIntToPtr(ptr, ptrTy);
 
     return ptr;
   };
 
-  auto streamOutBufDescTy = FixedVectorType::get(m_builder.getInt32Ty(), 4);
-  auto streamOutTableTy = ArrayType::get(streamOutBufDescTy, MaxTransformFeedbackBuffers);
-  auto streamOutTablePtrTy = PointerType::get(streamOutTableTy, ADDR_SPACE_CONST);
-
-  auto streamOutTablePtr = makePointer(streamOutTablePtrValue, streamOutTablePtrTy);
-
-  // NOTE: The stream-out control buffer has the following memory layout:
-  //   OFFSET[X]: OFFSET0, OFFSET1, ..., OFFSETN
-  //   FILLED_SIZE[X]: FILLED_SIZE0, FILLED_SIZE1, ..., FILLED_SIZEN
-  auto streamOutControlBufTy = ArrayType::get(ArrayType::get(m_builder.getInt32Ty(), MaxTransformFeedbackBuffers), 2);
-  auto streamOutControlBufPtrTy = PointerType::get(streamOutControlBufTy, ADDR_SPACE_CONST);
-
-  auto streamOutControlBufPtr = makePointer(streamOutControlBufPtrValue, streamOutControlBufPtrTy);
+  // NOTE: We simply represent the constant buffer with [256 x i32]. The actual size is smaller than 256 but it doesn't
+  // matter.
+  const auto constBufferPtrTy = PointerType::get(ArrayType::get(m_builder.getInt32Ty(), 256), ADDR_SPACE_CONST);
```

Comment:
This can be simplified due to opaque pointers. Just do `PointerType::get(ADDR_SPACE_CONST)`

Reply:
Will fix it.